From 5bc1cba2f5a3be41e76261f7c64cda995187a440 Mon Sep 17 00:00:00 2001
From: "Lin, Fanli" <fanli.lin@intel.com>
Date: Thu, 30 May 2024 20:37:21 -0700
Subject: [PATCH] add more not_device tests

---
 conftest.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/conftest.py b/conftest.py
index 3f2dae258..a2169833c 100644
--- a/conftest.py
+++ b/conftest.py
@@ -72,6 +72,9 @@ NOT_DEVICE_TESTS = {
     "/repo_utils/",
     "/utils/",
     "/agents/",
+    "/sagemaker/",
+    "/bettertransformer/",
+    "ModelOnTheFlyConversionTester"
 }
 
 # allow having multiple repository checkouts and not needing to remember to rerun
-- 
2.34.1

